homeassistant/components/iskra/__init__.py

```diff
@@ -0,0 +1,146 @@
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.entity import Entity
+
+from .const import DOMAIN, MANUFACTURER
+
+PLATFORMS: list[Platform] = [Platform.SENSOR]
+
+
+type IskraConfigEntry = ConfigEntry[Device]
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=60)
+TIME_TILL_UNAVAILABLE = timedelta(minutes=5)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: IskraConfigEntry) -> bool:
+    """Set up iskra device from a config entry."""
+    conf = entry.data
+    adapter = None
+
+    if conf[CONF_PROTOCOL] == "Modbus TCP":
+        adapter = Modbus(
+            ip_address=conf[CONF_HOST],
+            protocol="tcp",
+            port=conf[CONF_PORT],
+            modbus_address=conf[CONF_ADDRESS],
+        )
+    elif conf[CONF_PROTOCOL] == "Rest API":
+        authentication = None
+        if conf.get(CONF_USERNAME) or conf.get(CONF_PASSWORD):
+            authentication = {
+                "username": conf[CONF_USERNAME],
+                "password": conf[CONF_PASSWORD],
+            }
+        adapter = RestAPI(ip_address=conf[CONF_HOST], authentication=authentication)
+    else:
+        _LOGGER.error(
+            "Invalid protocol. Supported protocols are 'Modbus TCP' and 'Rest API'"
+        )
+        return False
+
+    try:
+        device = await Device.create_device(adapter)
+    except DeviceConnectionError as e:
+        _LOGGER.error("Cannot connect to the device: %s", e)
+        return False
+    except NotAuthorised as e:
+        _LOGGER.error("Not authorised: %s", e)
+        return False
+    except DeviceNotSupported as e:
+        _LOGGER.error("Device not supported: %s", e)
+        return False
+
+    await device.init()
+
+    entry.runtime_data = device
+
+    await hass.config_entries.async_forward_entry_setups(entry, PLATFORMS)
+
+    return True
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: IskraConfigEntry) -> bool:
+    """Unload a config entry."""
+    return await hass.config_entries.async_unload_platforms(entry, PLATFORMS)
+
+
+class IskraDevice(Entity):
+    """Representation a base Iskra device."""
+
+    _attr_should_poll = True
+
+    def __init__(self, device, gateway, config_entry):
+        """Initialize the Iskra device."""
+        self._state = None
+        self._is_available = True
+        self._serial = device.serial
+        self._model = device.model
+        self._fw_version = device.fw_version
+        self._device_name = f"{self._serial}"
+        self._remove_unavailability_tracker = None
+        self._device = device
+        self.gateway = gateway
+        self._gateway_id = config_entry.unique_id
+
+        self._is_gateway = self._device.is_gateway
+        self._device_id = self._serial
+
+    @property
+    def device_id(self):
+        """Return the device id of the Iskra device."""
+        return self._device_id
+
+    @property
+    def device_info(self) -> DeviceInfo:
```

Comment:
What entities does the gateway have?

Reply:
currently only number of connected devices, but i'will add more in follow up pull requests.
It also has pulse counter and temperature sensor user can use, tariff calculation and other status data.

Reply:
Can't we see the amount of connected devices via the HA?

Reply:
true we can, will delete this entity

Reply:
true we can, will delete this entity as it's unnecesary